flink-runtime/src/main/java/org/apache/flink/runtime/io/network/NetworkEnvironment.java

```diff
@@ -202,28 +233,97 @@ public TaskKvStateRegistry createKvStateTaskRegistry(JobID jobId, JobVertexID jo
 		return kvStateRegistry.createTaskRegistry(jobId, jobVertexId);
 	}
 
+	public ResultPartition[] getResultPartitionWriters(String taskId) {
+		return allPartitions.get(taskId);
+	}
+
+	public SingleInputGate[] getInputGates(String taskId) {
+		return allInputGates.get(taskId);
+	}
+
 	// --------------------------------------------------------------------------------------------
-	//  Task operations
+	//  Create Output Writers and Input Readers
 	// --------------------------------------------------------------------------------------------
 
-	public void registerTask(Task task) throws IOException {
-		final ResultPartition[] producedPartitions = task.getProducedPartitions();
+	public ResultPartitionWriter[] createResultPartitionWriters(
+			String taskId,
+			JobID jobId,
+			ExecutionAttemptID executionId,
```

Comment:
I think we could refactor the `ResultPartitionDeploymentDescriptor` to contain `ResultPartitionID` directly, then it might seem better to remove `ExecutionAttemptID` here as a final form of `ShuffleService#createResultPartitionWriter`.  I plan to do it in a separate issue.

Reply:
This issue is done by #7835